### ISSUE TITLE
Regression in v1.16.0-beta.2 - the server could not properly serve the CR schema

### ISSUE THREAD

**Issue by aanm** [kubernetes/kubernetes#82436]
**What happened**:
While upgrading from v1.15 to try out kubernetes v1.16.0-beta.2, kube-apiserver is printing error messages:
```
customresource_handler.go:307] the server could not properly serve the CR schema
```

This also **brakes** the API server since the user is unable to even delete the "faulty" CR

**What you expected to happen**:
- Not to print any error message and accept a CR that was already valid in 1.15

**How to reproduce it (as minimally and precisely as possible)**:

- Install k8s 1.15.x (optional)
- Create the following CR
- Upgrade to k8s v1.16.0-beta.2
```yaml
apiVersion: apiextensions.k8s.io/v1beta1
kind: CustomResourceDefinition
metadata:
  name: examples.example.com
spec:
  conversion:
    strategy: None
  group: example.com
  names:
    kind: Example
    listKind: ExampleList
    plural: examples
    singular: example
  preserveUnknownFields: true
  scope: Namespaced
  subresources:
    status: {}
  validation:
    openAPIV3Schema:
      properties:
        TypeFooBar:
          type: array
  version: v2
  versions:
  - name: v2
    served: true
    storage: true
status:
  acceptedNames:
    kind: Example
    listKind: ExampleList
    plural: examples
    singular: example
  conditions:
  - lastTransitionTime: "2019-09-06T17:43:58Z"
    message: no conflicts found
    reason: NoConflicts
    status: "True"
    type: NamesAccepted
  - lastTransitionTime: null
    message: the initial names have been accepted
    reason: InitialNamesAccepted
    status: "True"
    type: Established
  storedVersions:
  - v2

```

- check kube-apiserver for errors:
```
6195 customresource_handler.go:665] SchemaError(com.example.v2.Example.TypeFooBar): array should have exactly one sub-item
6195 customresource_handler.go:307] the server could not properly serve the CR schema
```
- Try delete the faulty CR (**Won't be possible**)

**Environment**:
- Kubernetes version (use `kubectl version`):
`v1.16.0-beta.2`

**Anything else we need to know?**:

A quick search sent me to this code path that is being used to compile k8s `v1.16.0-beta.2`
https://github.com/kubernetes/kubernetes/blob/6a11f9d46db8f4a627e4a59c61fca9c31c874298/vendor/k8s.io/kube-openapi/pkg/util/proto/document.go#L177-L179

It seems the problem it known but not yet fixed since this is in the latest commit of `k/kube-openapi`:

https://github.com/kubernetes/kube-openapi/blob/814a8073653e40e0e324205d093770d4e7bb811f/pkg/util/proto/document.go#L182-L186

/kind bug
/king regression

cc @sttts @nikhita

**Comment by liggitt** [comment]
This is caused by compiling the openapi spec for use by server-side apply here:

https://github.com/kubernetes/kubernetes/blob/38752f7f99869ed65fb44378360a517649dc2f83/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go#L648-L668

SSA is using the `ToProtoModels` method, which has significant gaps in handling valid openapi specs (see all the TODO comments in https://github.com/kubernetes/kubernetes/blob/master/vendor/k8s.io/kube-openapi/pkg/util/proto/document.go). Using it in the server-side path is concerning.

/assign @jennybuckley

**Comment by liggitt** [comment]
running with `--feature-gates=ServerSideApply=false`, this works as expected and is compatible with 1.15

**Comment by liggitt** [comment]
fix and covering test is in https://github.com/kubernetes/kubernetes/pull/82438

**Comment by liggitt** [comment]
this is resolved in [v1.16.0-rc.1](https://github.com/kubernetes/kubernetes/releases/tag/v1.16.0-rc.1), available now
